File: Frontend/Portal/src/app/models/agendamento.ts
import { addHours, startOfDay } from 'date-fns';

import { Procedimento } from './procedimento';
import { Cliente } from './cliente';

export class Agendamento {
    id?:number;
    dataInicio:Date = addHours(startOfDay(new Date()), 8);
    dataFim:Date = addHours(startOfDay(new Date()), 9);
    confirmado:boolean = false;
    pago:boolean = false;;
    valor?:number;
    custoMaterial?:number;
    custoFixo?:number;
    clienteId?:number;
    cliente?: Cliente;
    procedimentoId?:number
    procedimento?: Procedimento;
}

File: Backend/Backend.Domain/Entities/Agendamento.cs
using System.ComponentModel.DataAnnotations;
using Backend.Domain.Interfaces;
using Microsoft.EntityFrameworkCore;

namespace Backend.Domain.Entities
{
    public class Agendamento : IFilterUserBase
    {
        [Key]
        [Required]
        public int Id { get; set; }

        [Required]
        public int UsuarioId { get; set; }

        public DateTime DataInicio { get; set; }
        
        public DateTime DataFim { get; set; }

        public bool Confirmado { get; set; } = false;
        
        public bool Pago { get; set; } = false;

        [Precision(8, 2)]
        public decimal Valor { get; set; }

        [Precision(8, 2)]
        public decimal CustoMaterial { get; set; }

        [Precision(8, 2)]
        public decimal CustoFixo { get; set; }

        [Required]
        public int ClienteId { get; set; }

        [Required]
        public int ProcedimentoId { get; set; }

        public virtual Cliente? Cliente { get; set; }
        
        public virtual Procedimento? Procedimento { get; set; }
    }
}
